ros_buildfarm/templates/snippet/scm_git.xml.em

```diff
@@ -18,6 +18,7 @@
     </branches>
     <doGenerateSubmoduleConfigurations>false</doGenerateSubmoduleConfigurations>
     <submoduleCfg class="list"/>
+    <recursiveSubmodules>true</recursiveSubmodules>
```

Comment:
As far as I can see this is not a valid configuration option. I think it needs to be the following instead (after the `hudson.plugins.git.extensions.impl.CloneOption` / `hudson.plugins.git.extensions.impl.PreBuildMerge` tag):

```
<hudson.plugins.git.extensions.impl.SubmoduleOption>
  <disableSubmodules>false</disableSubmodules>
  <recursiveSubmodules>true</recursiveSubmodules>
  <trackingSubmodules>false</trackingSubmodules>
  <reference></reference>
  <parentCredentials>false</parentCredentials>
  </hudson.plugins.git.extensions.impl.SubmoduleOption>
</extensions>
```

Reply:
Ok, from the test output it just looked like it was already cloning with `--recurse-submodules`, although I have to admit that I don't really know where to look at exactly.
Also I couldn't really find a reference for specifying these options in such an XML config file, only how to set it in the web interface.

Reply:
Ah, thx... you already updated it.

Reply:
> Ok, from the test output it just looked like it was already cloning with `--recurse-submodules`.

The Travis CI won't tell anything about Jenkins. Those don't run a Jenkins server but only invoke the "script" versions of all jobs. They only show that your change in the `devel_script_clone.sh.em` file is being used and works.

Reply:
Ah, that makes sense.

Reply:
I think this added line still needs to be reverted. The newly added block below is performing the intended feature.